arbstate/daprovider/util.go

```diff
@@ -95,6 +95,10 @@ const KnownHeaderBits byte = DASMessageHeaderFlag | TreeDASMessageHeaderFlag | L
 
 // hasBits returns true if `checking` has all `bits`
 func hasBits(checking byte, bits byte) bool {
+	if bits == EigenDAMessageHeaderFlag && checking != EigenDAMessageHeaderFlag {
+		return false
+	}
```

Comment:
prob want to add comment mentioning the bug and what exactly this is doing in english. Even with the image from the COE I'm not quite sure I understand the full context.
<img width="781" height="269" alt="image" src="https://github.com/user-attachments/assets/26c3833b-00da-4089-b9b6-107bccf4dca4" />

This looks like a major design flaw that should be fixed somewhere else (perhaps EigenDAMessageHeaderFlag needs to change and shouldnt have the first bit set to 1?
Or Anytrust should not be 0x80?